genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/v4/converters/JobServiceProtoConverter.java

```diff
@@ -249,6 +250,7 @@ public ChangeJobStatusRequest toProtoChangeJobStatusRequest(
         builder.putAllEnvironmentVariables(jobSpecification.getEnvironmentVariables());
         builder.setIsInteractive(jobSpecification.isInteractive());
         builder.setJobDirectoryLocation(jobSpecification.getJobDirectoryLocation().getAbsolutePath());
+        builder.setArchiveLocation(jobSpecification.getArchiveLocation());
```

Comment:
This will throw if the argument is null.

Reply:
I was converting to an empty value in the jobSpecification constructor to handle that. 
Will be moving to a getter which returns Optional<String> for archiveLocation and will do the necessary check before populating.

Reply:
I think the nuances of dealing with proto fields belong in the proto converter